start-client/src/components/common/share/SharePopup.js

```diff
@@ -0,0 +1,129 @@
+    this.handleClickOutside = this.handleClickOutside.bind(this)
+  }
+
+  componentDidMount = () => {
+    document.addEventListener('mousedown', this.handleClickOutside)
+  }
+
+  componentWillUnmount = () => {
+    document.removeEventListener('mousedown', this.handleClickOutside)
+  }
+
+  setWrapperRef = node => {
+    this.wrapperRef = node
+  }
+
+  handleClickOutside = event => {
+    if (this.wrapperRef && !this.wrapperRef.contains(event.target)) {
+      this.setOnClose()
+    }
+  }
+
+  setOnClose = () => {
+    this.props.onClose()
+  }
+
+  getUrl = () => {
+    const rootValues = this.props.rootValues
+    const values = this.props.properties
+    return getShareUrl(values, rootValues)
+  }
+
+  onCopy = () => {
+    toast.success('Your URL has been copied in the clipboard with success.')
+    this.props.onClose()
+  }
+
+  shareOnTwitter = () => {
+    const textParam = querystring.stringify({
+      text: `Sharing my configuration ${this.getUrl()}`,
+    })
+    const urlTwitter = `https://twitter.com/intent/tweet?${textParam}`
+    window.open(urlTwitter, '_blank')
+    this.props.onClose()
+  }
+
+  render = () => {
+    const urlToShare = this.getUrl()
+    return (
+      <TransitionGroup component={null}>
+        {this.props.open && (
+          <CSSTransition classNames='popup' timeout={500}>
+            <div className='popup-share'>
+              <div ref={this.setWrapperRef}>
+                <div className='popup-header'>
+                  <h1>Share your project</h1>
+                  <a
+                    href='/#'
+                    onClick={e => {
+                      e.preventDefault()
+                      this.setOnClose()
+                    }}
+                    className='close'
+                  >
+                    <IconTimes />
+                  </a>
+                </div>
+                <div className='popup-content'>
+                  <div>
+                    <label>Link to your configuration</label>
+                    <CopyToClipboard onCopy={this.onCopy} text={urlToShare}>
+                      <a
+                        href='/#'
+                        onClick={e => {
+                          e.preventDefault()
+                        }}
+                        className='button button-default'
+                      >
+                        <IconCopy />
+                        Copy the URL
+                      </a>
+                    </CopyToClipboard>
+                    <label className='label-padding-top'>
+                      Share on social network
+                    </label>
+                    <a
+                      href='/#'
+                      onClick={e => {
+                        e.preventDefault()
+                        this.shareOnTwitter()
+                      }}
+                      className='button button-default'
+                    >
+                      <IconTwitter />
+                      Share on twitter
```

Comment:
I don't think we need to provide the ability to share directly to Twitter. IMO, copying the URL to the clipboard is sufficient. It can then be pasted wherever the user wants.